basictracer/text_propagator.py

```diff
@@ -54,7 +54,10 @@ def inject(self, span_context, carrier):
             for k in span_context.baggage:
                 carrier[prefix_baggage+k] = span_context.baggage[k]
 
-    def extract(self, carrier):  # noqa
+    def extract(self, carrier):
+        return self._extract(carrier)
+
+    def _extract(self, carrier, allow_zero_count=False):  # noqa
```

Comment:
I agree with your PR description - no incoming trace context should result in None. I don't think `allow_zero_count` customization is necessary.

Reply:
That was my original thought but there is a test that [explicitly checks this case](https://github.com/opentracing/basictracer-python/blob/9618591fcd6b34e12574778c2ef1e0d9bb9d8164/tests/test_propagation.py#L73), which is when I went for the HTTP specific approach. I'm happy either way.

Reply:
yeah, I think it's ok to change that test. I think what it should be checking is if N of fields is 0 < N < 3, then we expect an error because the headers are malformed, but if it's ==0 then it's ok.

Reply:
btw, please add a test for ==0 case.